FanartHandler/FanartSelectedOther.cs

```diff
@@ -850,6 +855,38 @@ public void AddSelectedMoviePropertys()
         strIMDBID = Utils.GetProperty("#myfilms.db.imdb_id.value");
       }
 
+      var window = GUIWindowManager.GetWindow(Utils.iActiveWindow);
+      GUIListItem selectedListItem = null;
+      if (window != null)
+      {
+        selectedListItem = GUIControl.GetSelectedListItem(Utils.iActiveWindow, window.GetFocusControlId());
+        if (selectedListItem == null && string.IsNullOrEmpty(strIMDBID))
+          return;
+      }
+
+      if (string.IsNullOrEmpty(strIMDBID))
+      {
+        strIMDBID = Utils.GetProperty("#Trakt.Movie.ImdbId");
+      }
+
+      string strTVDBID = Utils.GetProperty("#Trakt.Show.TvdbId");
+      if (string.IsNullOrEmpty(strTVDBID))
+      {
+        strTVDBID = Utils.GetProperty("#TVSeries.Series.ID");
+      }
+      if (string.IsNullOrEmpty(strTVDBID))
+      {
+        // TV Series ID
+        if (selectedListItem != null)
+        {
+          DBSeries series = selectedListItem.TVTag as DBSeries;
+          if (series != null)
+          {
+            strTVDBID = series[DBOnlineSeries.cID];
+          }
+        }
+      }

```

Comment:
I looked at it with fresh eyes, after sleep. And somehow I don’t like this code. It will always work (on all plugins, and even without the TV Series plugin installed, and this can lead to the collapse of the logic or plugin.

I propose to redo this. Add the method for obtaining TV Series ID to the TV Series module. And add analysis to the active window.

https://github.com/andrewjswan/mediaportal-fanart-handler/blob/master/FanartHandler/UtilsTVSeries.cs

Reply:
For example add a method:
```C#
    internal static string GetTVSeriesID(int ControlID)  // -> TV Series ID ...
    {
      if (!Utils.TVSeriesEnabled)
      {
        return string.Empty;
      }
      if (Utils.ActiveWindow == TV Series Window ID)
      {
        return string.Empty;
      }
      GUIWindow window = GUIWindowManager.GetWindow(Utils.iActiveWindow);
      GUIControl cntl = window.GetControl(50);
      if (cntl != null && cntl is GUIFacadeControl)
      {
         GUIListItem selectedListItem = (control as GUIFacadeControl).SelectedListItem;
         if (selectedListItem != null)
         {
           DBSeries series = selectedListItem.TVTag as DBSeries;
           if (series != null)
           {
             return series[DBOnlineSeries.cID];
           }
         }
      }
        return string.Empty;
    }
```
And call this in the block:
https://github.com/andrewjswan/mediaportal-fanart-handler/blob/4d1e117d1046b6fd88977373f195736f387a951f/FanartHandler/FanartSelectedOther.cs#L879-L887

Reply:
`if (Utils.iActiveWindow == TVSeriesWindowID)`
Where do I get the TVSeriesWindowID?

How do I call the control id?
`strTVDBID = UtilsTVSeries.GetTVSeriesID(window.GetFocusControlId());`

Reply:
https://github.com/andrewjswan/mediaportal-fanart-handler/blob/2bfcc82f298900cf70fc6a1fded48332f9411e32/FanartHandler/Utils.cs#L2576-L2577

Reply:
> How do I call the control id?
> `strTVDBID = UtilsTVSeries.GetTVSeriesID(window.GetFocusControlId());`

`strTVDBID = UtilsTVSeries.GetTVSeriesID(50);`
Facade Control in TV Series has an ID - 50.

Reply:
> https://github.com/andrewjswan/mediaportal-fanart-handler/blob/2bfcc82f298900cf70fc6a1fded48332f9411e32/FanartHandler/Utils.cs#L2576-L2577

Makes no sense to do that. That makes it work for Trakt series but not for TVSeries plugin.

Reply:
> Makes no sense to do that. That makes it work for Trakt series but not for TVSeries plugin. Why do we need this when it works without this check?

Here we only get IDs for Series, and this will not work with Trakt.
https://github.com/andrewjswan/mediaportal-fanart-handler/blob/4d1e117d1046b6fd88977373f195736f387a951f/FanartHandler/FanartSelectedOther.cs#L879-L887

Reply:
> Here we only get IDs for Series, and this will not work with Trakt.

#Trakt.Show.TvdbId is read by Trakt for TVSeries in the Trakt plugin. In the TVSeries plugin this is required instead as Trakt is empty `strTVDBID = UtilsTVSeries.GetTVSeriesID(50);`
But you then tell it to return null. I think you meant to make it check if it's not equal?

Reply:
Change to this?
```
      if (Utils.iActiveWindow != 9811 &&    // TVSeries
          Utils.iActiveWindow != 9813)      // TVSeries Playlist
```

Reply:
I've lost track of what we're discussing. 
That part of the code is called for all plugins, since we need to get something from a specific plugin, and this is not a property (but we need to analyze data from the plugin), then it’s worth checking whether its window is active, whether it is installed, and other checks. And it’s better to do this in a helper.

Reply:
There are real stabiliity issues with the DBOnlineSeries.cID check method. I added a `<!-- #TVSeries.Series.ID -->` placeholder to TVSeries.xml 9811 and TVSeries.Playlist.xml 9813 and it's very stable with no issues. I will push the change to MePo theme and comment out the other method.